lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+  Checkbox,
+  CheckboxCheckedFilledIcon,
+  Input,
+  RadioGroup,
+} from '@hypothesis/frontend-shared';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useId } from 'preact/hooks';
+
+export type GradingType = 'all_or_nothing' | 'scaled';
+
+export type AutoGradingConfig = {
+  /** Whether auto grading is active for the assignment or not */
+  active?: boolean;
+
+  /**
+   * - all_or_nothing: students need to meet a minimum value, making them get
+   *                   either 0% or 100%
+   * - scaled: students may get a proportional grade based on the amount of
+   *           annotations. If requirement is 4, and they created 3, they'll
+   *           get a 75%
+   */
+  gradingType: GradingType;
+
+  /**
+   * - cumulative: both annotations and replies will be counted together for
+   *               the grade calculation
+   * - separately: students will have different annotation and reply goals.
+   */
+  activityCalculation: 'cumulative' | 'separately';
+
+  /**
+   * Required number of annotations if activityCalculation is 'separately' or
+   * annotations+replies otherwise
+   */
+  requiredAnnotations: number;
+
+  /**
+   * Required number of replies if activityCalculation is 'separately'
+   */
+  requiredReplies?: number;
+};
+
+type AnnotationsGoalInputGroupProps = {
+  children?: ComponentChildren;
+  gradingType: GradingType;
+  value: number;
+  onChange: (newValue: number) => void;
+
+  /** Minimum required value for the input. Defaults to 1 */
+  min?: number;
+};
+
+/**
+ * Controls containing a number input to set the amount of required annotations
+ * or replies
+ */
+function AnnotationsGoalInputGroup({
```

Comment:
Yeah, I see the fact that `Input` and `Group` are together (`InputGroup`) is confusing. What I meant is "this is a group of elements which includes an input used to define the annotations goal".

By looking at the structure it has now (one label and one input) perhaps it could simply be `AnnotationsGoalInput`. At some point it was more complex, but I ended up simplifying it after a few rounds.